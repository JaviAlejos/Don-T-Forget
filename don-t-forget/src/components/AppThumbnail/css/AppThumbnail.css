.AppThumbnailCommon {
  margin-bottom: 0px;
  margin-top: 0;
  padding: 0;
}

.AppThumbnailImage {
  margin-left: 100px;
}

.AppThumbnail{
  margin-left: 70px;
  width: 45%;
}

.AppThumbnailInput {
  width: 70px;
}

.AppThumbnailButton {
  margin-left: 105px;
}
