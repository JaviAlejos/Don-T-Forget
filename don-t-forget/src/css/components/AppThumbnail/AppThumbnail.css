.AppThumbnail {
  margin-left: 50px;
  margin-top: 0;
  padding: 0;
  width: 45%;
}

.AppThumbnailInput {
  width: 70px;
}

.AppThumbnailButton {
  margin-left: 70px;
  margin-top: 0;
  padding: 0;
}

.AppThumbnailButtonAdd {
  margin-left: 35px;
  margin-top: 0;
  padding: 0;
}

.AppThumbnailRadio{
  margin-left: 50%;
  margin-top: 1%;
  padding: 0;
}
