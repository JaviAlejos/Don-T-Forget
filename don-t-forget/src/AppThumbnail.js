import React, {Component} from 'react';
import {Thumbnail,Button,OverlayTrigger,Tooltip} from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.css';
import AppPassword from './AppPassword';
import './css/components/AppThumbnail/AppThumbnail.css';


class AppThumbnail extends Component {

    render() {
      const {show,value,name}=this.props;
      const tooltip = (<Tooltip id="tooltip"><strong>{name}</strong></Tooltip>);

      return (
          <OverlayTrigger placement="top" overlay={tooltip}>
            <Thumbnail src={`../icons/${name}.png`} alt="70x70">
                <p>
                  <AppPassword value={value} show={show} className="AppThumbnail"/>
                </p>
                <Button bsStyle="primary" bsSize="small" className="AppThumbnailButton">Save</Button>

              </Thumbnail>
          </OverlayTrigger>

              /*<Thumbnail src="../icons/Plus.png" alt="by tFity">
              GitHub --> by Bo-Yi Wu
              Twitter --> by lonaug
              Facebook $$ Google Instagram --> by wikimedia commons $$ YaNiS2017

              https://facebook.github.io/react/docs/forms.html--> The select Tag
              */

        );
    }
}

export default AppThumbnail;
