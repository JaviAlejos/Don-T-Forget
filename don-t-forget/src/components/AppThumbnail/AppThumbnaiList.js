import React, {Component} from 'react';
import {Grid,Row,Col} from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.css';
import './css/AppThumbnail.css';
import AppThumbnail from './AppThumbnail';
import AppThumbnailForm from './AppThumbnailForm';
import {connect} from 'react-redux';


class AppThumbnaiList extends Component {

    render() {
      return (
        <Grid>
          <Row>
            {
                this.props.passwords.map(password=>{
                      return (
                        <Col xs={4} md={4}>
                          <AppThumbnail name={password.namePass} />
                        </Col>
                  );
                })}

            <Col xs={4} md={4}>
              <AppThumbnailForm/>
            </Col>
          </Row>
          </Grid>
        );
    }
}

var mapStateToProps=function(state) {
  return {
    passwords:state.passwords
  }
}

export default connect(mapStateToProps)(AppThumbnaiList);
