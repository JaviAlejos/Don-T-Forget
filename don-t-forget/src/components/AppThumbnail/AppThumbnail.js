import React, {Component} from 'react';
import {Thumbnail,OverlayTrigger,Tooltip,Glyphicon} from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.css';
import '../AppPassword/AppPassword.css';
import './css/AppThumbnail.css';
import {connect} from 'react-redux';
import firebase from 'firebase';
import AppModalThumbnail from './AppModalThumbnail';




class AppThumbnail extends Component {

  constructor(props) {
     super(props);
     this.deletePassword = this.deletePassword.bind(this);
     this.state = {value: this.props.name,showModal: false};
     this.closeAppModal = this.closeAppModal.bind(this);
     this.showAppModal = this.showAppModal.bind(this);
  }

  closeAppModal() {
      this.setState({value:this.state.value,showModal: false});
  }

  showAppModal() {
      this.setState({value:this.state.value,showModal: true});
  }


deletePassword(){
  const {name,deletePass,passwords}=this.props;
  const erase = confirm("Are you sure?");
    if (erase)
      deletePass({namePass:name});

    //delete from firebase database
      const ref = firebase.database().ref(`passwords/${passwords.filter(pass => pass.namePass==name)[0].idPassword}`);
      ref.remove();
}


render() {

  const {name}=this.props;
  const names=["Twitter","GitHub","Instagram","Facebook","Google"];
  let varname="Undefined";

  if (names.indexOf(name)!=-1)
      varname=name;

      const tooltipName = (<Tooltip id="tooltip"><strong>{name}</strong></Tooltip>);
      const tooltipEdit = (<Tooltip id="tooltip"><strong>Edit Password</strong></Tooltip>);
      const tooltipDelete = (<Tooltip id="tooltip"><strong>Delete Password</strong></Tooltip>);
      const tooltipSend = (<Tooltip id="tooltip"><strong>Send an email with the password</strong></Tooltip>);



      return (
          <Thumbnail >
                          <AppModalThumbnail showModalDialog={this.state} close={this.closeAppModal} add={false}/>
              <p>
                <OverlayTrigger placement="top" overlay={tooltipName}>
                  <img src={`../icons/${varname}.png`} alt="70x70" className="AppThumbnailImage"/>
                </OverlayTrigger>
              </p>
              <OverlayTrigger placement="top" overlay={tooltipSend}>
                <span>
                  <Glyphicon className="AppThumbnailButton GlyphiconOk" glyph="glyphicon glyphicon-envelope" />
                </span>
              </OverlayTrigger>

              <OverlayTrigger placement="top" overlay={tooltipEdit}>
                <span onClick={this.showAppModal}>
                  <Glyphicon className="StandardComponent GlyphiconWarning" glyph="glyphicon glyphicon-pencil" />
                </span>
              </OverlayTrigger>

              <OverlayTrigger placement="top" overlay={tooltipDelete}>
                <span onClick={this.deletePassword}>
                  <Glyphicon className="StandardComponent GlyphiconDanger" glyph="glyphicon glyphicon-remove" />
                </span>
              </OverlayTrigger>

          </Thumbnail>

              /*<Thumbnail src="../icons/Plus.png" alt="by tFity">
              GitHub --> by Bo-Yi Wu
              Twitter --> by lonaug
              Facebook $$ Google Instagram --> by wikimedia commons $$ YaNiS2017

              https://facebook.github.io/react/docs/forms.html--> The select Tag
              */

        );

    }
}

var mapStateToProps = function(state) {
    return {passwords: state.passwords}
}

var mapDispatchToProps = function(dispatch) {
  return {
      deletePass: function(password) {
        dispatch({
          type: 'DELETE_PASSWORD',
          password
        })}
    }
}

export default connect(mapStateToProps,mapDispatchToProps)(AppThumbnail);
