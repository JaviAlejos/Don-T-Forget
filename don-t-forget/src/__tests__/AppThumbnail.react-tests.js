//importas react
import React from 'react';
//importas el componente
import AppThumbnail from '../components/AppThumbnail/AppThumbnail';
//importas el test-render
import renderer from 'react-test-renderer';

import { createStore } from 'redux';
import { Provider } from 'react-redux';

var passwords=function (state=[], action) {
  debugger;
  switch (action.type) {
  case 'ADD_PASSWORD':
      const exist = state.findIndex(pass => pass.namePass === action.password.namePass);
      if(exist<0)
        return state.concat([ action.password ]);
        else
          return state;
  case 'DELETE_PASSWORD':
    return state.filter(pass => pass.namePass!=action.password.namePass);

  case 'DELETE_PASSWORDS':
    return [];
  default:
    return state;
  }
}


test(
    'Simple Google Thumbnail', //titulo del test
    //funcion del test
    () => {
      const store = createStore(passwords) // can also be a mock
    const component = renderer.create(


       <Provider store={store}>
         <AppThumbnail name={"Google"}/>
       </Provider>


  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

/*example*************

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();*/
}

);
