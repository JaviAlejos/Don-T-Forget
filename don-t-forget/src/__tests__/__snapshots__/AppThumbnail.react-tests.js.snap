// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Google Thumbnail 1`] = `
<div
  className="thumbnail"
>
  <img
    alt={undefined}
    src={undefined}
  />
  <div
    className="caption"
  >
    <p>
      <img
        alt="70x70"
        className="AppThumbnailImage"
        onBlur={[Function]}
        onClick={null}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        src="../icons/Google.png"
      />
    </p>
    <span
      onBlur={[Function]}
      onClick={null}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <span
        className="AppThumbnailButton GlyphiconOk glyphicon glyphicon-glyphicon glyphicon-envelope"
      />
    </span>
    <span
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <span
        className="StandardComponent GlyphiconWarning glyphicon glyphicon-glyphicon glyphicon-pencil"
      />
    </span>
    <span
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <span
        className="StandardComponent GlyphiconDanger glyphicon glyphicon-glyphicon glyphicon-remove"
      />
    </span>
  </div>
</div>
`;
